aten/src/ATen/native/mkldnn/Conv.cpp

```diff
@@ -215,6 +315,27 @@ Tensor _mkldnn_convolution(
   const ideep::tensor x = itensor_from_tensor(input);
   const ideep::tensor w = itensor_from_tensor(weight);
 
+  const std::vector<int64_t> strides = {stride.begin(), stride.end()};
+  const std::vector<int64_t> dilates = {dilation.begin(), dilation.end()};
+  const std::vector<int64_t> paddings = {padding.begin(), padding.end()};
+
+  // Look up in cache of packed weight
+  void* w_buf_ptr = w.get_data_handle();
+  auto sample1 = w.data_type_size(w.get_data_type()) == sizeof(float) ?
+                 reinterpret_cast<int32_t*>(w_buf_ptr)[0] :
+                 (int32_t)reinterpret_cast<int16_t*>(w_buf_ptr)[0];
+  auto sample2 = w.data_type_size(w.get_data_type()) == sizeof(float) ?
+                 reinterpret_cast<int32_t*>(w_buf_ptr)[weight_t.numel()-1] :
+                 (int32_t)reinterpret_cast<int16_t*>(w_buf_ptr)[weight_t.numel()-1];
+  std::vector<int32_t> weight_data_samples = {sample1, sample2};
+  auto key = create_weight_cache_key(
+      x.get_dims(), w.get_dims(), w_buf_ptr, weight_data_samples,
+      strides, dilates, paddings, paddings, groups, op_attr, memory_format,
+      input_t.scalar_type(), weight_t.scalar_type(), at::get_num_threads());
+  const ideep::tensor& expected_w =
+      get_expected_weight(key, x, w, strides, paddings, paddings,
+                          dilates, groups, op_attr, is_channels_last);
```

Comment:
Would this cause problem if the weight is changed (not constant)? What is the assumption you make about it here?

Reply:
This is not the right thing to do - either check weight content or check key only. We don't have a reliable way of knowing if weight is constant without the caller telling us.

Reply:
We have made weight info part of the key. We find packed weight in cache by key. The key is an `std::string` consisting of various info, including input/weight shape, weight buffer address, weight data samples, number of threads, etc. If weight is changed, the key is also changed.

Reply:
Well, you know that's not enough, right? Users can change the weight content from the frontend without changing the key you define here.

Reply:
OK, I will check all elements of the weight tensor. But I think the overhead may be too big to create the key. I can implement it this way first and collect some perf data to validate.

Reply:
No, I don't think scanning weights is the right thing to do.